src/App.jsx

```diff
@@ -48,8 +50,23 @@ function App(props) {
     <Router>
       <div>
         {/* Protected Routes */}
-        <ProtectedRoute exact path="/" component={Home} />
+        <ProtectedRoute exact path="/" component={Home} view={"home"} />
         <ProtectedRoute path="/create-post" component={CreatePost} />
+        <ProtectedRoute
+          path="/side-bar"
+          component={Sidebar}
+          logout={props.logoutThunk}
+        />
+        <ProtectedRoute
+          path="/:username-posts"
+          component={Home}
+          view={"userPosts"}
+        />
+        <ProtectedRoute
+          path="/:username-vote-history"
+          component={Home}
+          view={"voteHistory"}
```

Comment:
This is not a good idea, to render the same component for multiple endpoints and pass in a view to differentiate.
You are better off having separate `containers` for each of these since they are separate endpoints.
Then, if you think there is a lot of re-usable code between them, you can move that code to a `component` and then share it. 
That is the whole point of re-usability and component architecture of React.
Otherwise, you will end up with a lot of if-else-if statements and have one massive component that does too many things

Reply:
Ok thats fair, refactoring this